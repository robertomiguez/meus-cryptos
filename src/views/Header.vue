<template>
  <div>
    <header class="header">
      <a href="" class="title">Meus Cryptos</a>
      <img class="logo" src="@/assets/squirrel.png">
      <input class="menu-btn" type="checkbox" id="menu-btn" />
      <label class="menu-icon" for="menu-btn"><span class="navicon"></span></label>
      <ul class="menu">
        <li><router-link to='/'>My positions</router-link></li>
        <li><router-link to='/allcoins'>All coins</router-link></li>
        <li><router-link to='/supportedcryptos'>Supported Cryptos</router-link></li>
        <li><router-link to='/howto'>Help ?</router-link></li>
      </ul>
    </header>
  </div>
</template>

<script>
export default {
  name: 'Header'
}
</script>

<style scoped>

a {
  color: ghostwhite;
}

/* header */

.header {
  box-shadow: 1px 1px 4px 0 rgba(0,0,0,.1);
  position: fixed;
  width: 100%;
  z-index: 3;
}

.header ul {
  margin: 0;
  padding: 0;
  list-style: none;
  overflow: hidden;
}

.header li a {
  display: block;
  padding: 20px 20px;
  border-right: 1px solid #f4f4f4;
  text-decoration: none;
}

.header li a:hover,
.header .menu-btn:hover {
  background-color: #333;
}

.header .title {
  display: block;
  float: left;
  font-size: 1.5em;
  padding: 18px 10px;
  text-decoration: none;
}

.header .logo {
  display: block;
  float: left;
  padding: 13px 0px;
  width: 2em;
}

/* menu */

.header .menu {
  clear: both;
  max-height: 0;
  transition: max-height .2s ease-out;
}

/* menu icon */

.header .menu-icon {
  cursor: pointer;
  display: inline-block;
  float: right;
  padding: 28px 20px;
  position: relative;
  user-select: none;
}

.header .menu-icon .navicon {
  background: #333;
  display: block;
  height: 2px;
  position: relative;
  transition: background .2s ease-out;
  width: 18px;
}

.header .menu-icon .navicon:before,
.header .menu-icon .navicon:after {
  background: #333;
  content: '';
  display: block;
  height: 100%;
  position: absolute;
  transition: all .2s ease-out;
  width: 100%;
}

.header .menu-icon .navicon:before {
  top: 5px;
}

.header .menu-icon .navicon:after {
  top: -5px;
}

/* menu btn */

.header .menu-btn {
  display: none;
}

.header .menu-btn:checked ~ .menu {
  max-height: 240px;
}

.header .menu-btn:checked ~ .menu-icon .navicon {
  background: transparent;
}

.header .menu-btn:checked ~ .menu-icon .navicon:before {
  transform: rotate(-45deg);
}

.header .menu-btn:checked ~ .menu-icon .navicon:after {
  transform: rotate(45deg);
}

.header .menu-btn:checked ~ .menu-icon:not(.steps) .navicon:before,
.header .menu-btn:checked ~ .menu-icon:not(.steps) .navicon:after {
  top: 0;
}

/* 48em = 768px */

@media (min-width: 48em) {
  .header li {
    float: left;
  }
  .header li a {
    padding: 20px 30px;
  }
  .header .menu {
    clear: none;
    float: right;
    max-height: none;
  }
  .header .menu-icon {
    display: none;
  }
  .header .title {
    font-size: 2em;
    padding: 13px 10px;
  }
  .header .logo {
    width: 3em;
    padding: 5px 0px;
  }
}

</style>
