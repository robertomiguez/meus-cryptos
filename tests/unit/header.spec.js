import { shallowMount } from '@vue/test-utils'
import Header from '@/views/Header'

describe('Header', () => {
  const build = () => {
    const wrapper = shallowMount(Header, {
      stubs: ['router-link', 'router-view']
    })

    return {
      wrapper
    }
  }

  it('renders the component', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
