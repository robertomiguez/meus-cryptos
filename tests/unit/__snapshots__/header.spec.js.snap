// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the component 1`] = `
<div>
  <header class="header"><a href="" class="title">Meus Cryptos</a> <img src="@/assets/squirrel.png" class="logo"> <input type="checkbox" id="menu-btn" class="menu-btn"> <label for="menu-btn" class="menu-icon"><span class="navicon"></span></label>
    <ul class="menu">
      <li>
        <router-link-stub to="/">My positions</router-link-stub>
      </li>
      <li>
        <router-link-stub to="/allcoins">All coins</router-link-stub>
      </li>
      <li>
        <router-link-stub to="/supportedcryptos">Supported Cryptos</router-link-stub>
      </li>
      <li>
        <router-link-stub to="/howto">Help ?</router-link-stub>
      </li>
    </ul>
  </header>
</div>
`;
